Userland/Libraries/LibEDID/EDID.cpp

```diff
@@ -0,0 +1,1189 @@
+
+        return checksum == 0;
+    };
+
+    size_t offset = sizeof(Definitions::EDID);
+    auto* raw_extension_blocks = (Definitions::ExtensionBlock const*)(m_bytes.data() + offset);
+    Definitions::ExtensionBlock const* current_extension_map = nullptr;
+    if (m_revision <= 3) {
+        if (raw_extension_block_count > 1) {
+            current_extension_map = &raw_extension_blocks[0];
+            if (read_host(&current_extension_map->tag) != Definitions::ExtensionBlockTag::ExtensionBlockMap)
+                return Error::from_string_literal("Did not find extension map at block 1"sv);
+            if (!validate_block_checksum(*current_extension_map))
+                return Error::from_string_literal("Extension block map checksum mismatch"sv);
+        }
+    } else if (read_host(&raw_extension_blocks[0].tag) == Definitions::ExtensionBlockTag::ExtensionBlockMap) {
+        current_extension_map = &raw_extension_blocks[0];
+    }
+
+    for (unsigned raw_index = 0; raw_index < raw_extension_block_count; raw_index++) {
+        auto& raw_block = raw_extension_blocks[raw_index];
+        u8 tag = read_host(&raw_block.tag);
+        if (current_extension_map && raw_index == 127) {
+            if (tag != Definitions::ExtensionBlockTag::ExtensionBlockMap)
+                return Error::from_string_literal("Did not find extension map at block 128"sv);
+            current_extension_map = &raw_extension_blocks[127];
+            if (!validate_block_checksum(*current_extension_map))
+                return Error::from_string_literal("Extension block map checksum mismatch"sv);
+            continue;
+        }
+
+        if (tag == Definitions::ExtensionBlockTag::ExtensionBlockMap)
+            return Error::from_string_literal("Unexpected extension map encountered"sv);
+
+        if (!validate_block_checksum(raw_block))
+            return Error::from_string_literal("Extension block checksum mismatch"sv);
+
+        IterationDecision decision = callback(raw_index + 1, tag, raw_block.block.revision, m_bytes.slice(offset, sizeof(Definitions::ExtensionBlock)));
+        if (decision != IterationDecision::Continue)
+            return decision;
+
+        offset += sizeof(Definitions::ExtensionBlock);
+    }
+    return IterationDecision::Continue;
+}
+
+String EDID::version() const
+{
+    return String::formatted("1.{}", (int)m_revision);
+}
+
+String EDID::legacy_manufacturer_id() const
+{
+    u16 packed_id = read_be(&raw_edid().vendor.manufacturer_id);
+    char id[4] = {
+        (char)((u16)'A' + ((packed_id >> 10) & 0x1f) - 1),
+        (char)((u16)'A' + ((packed_id >> 5) & 0x1f) - 1),
+        (char)((u16)'A' + (packed_id & 0x1f) - 1),
+        '\0'
+    };
+    return id;
+}
+
+u16 EDID::product_code() const
+{
+    return read_le(&raw_edid().vendor.product_code);
+}
+
+u32 EDID::serial_number() const
+{
+    return read_le(&raw_edid().vendor.serial_number);
+}
+
+auto EDID::digital_display() const -> Optional<DigitalDisplay>
```

Comment:
Same here, and below. I might be missing something, but mixing both styles seems odd :)